src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstance.cs

```diff
@@ -354,7 +361,12 @@ public TelemetryVerifier(VisualStudioInstance instance)
             /// <param name="expectedEventNames"></param>
             public void VerifyFired(params string[] expectedEventNames)
             {
-                _instance.WaitForTelemetryEvents(expectedEventNames);
+                var telemetryEnabled = _instance.TryWaitForTelemetryEvents(expectedEventNames);
+                if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("AGENT_ID")))
```

Comment:
I think it might be clearer if the build script set an environment variable when the `-ci` option is passed instead of relying on an AzDO variable being present.

Reply:
➡️ Fixed